DESIGN.md

```diff
@@ -0,0 +1,108 @@
+
+## Vanagon for the Windows platform
+
+### Overview
+Vanagon was adapted for the windows platform using WiX. At a very high level each component of the project gets built on a build system, then gets moved to a staging directory where we use the WiX toolset to build an MSI. This workflow is consistent with the rest of vanagon, however the WiX toolset is not so simple that there are no caveats, specifically:
+
+* Vanagon does not contain functionality to handle registry searching
+* Vanagon does not contain functionality to handle the MSI UI
+* If the project requires MSI custom actions (such as those to search the registry, set properties used in the MSI etc.) Vanagon has no functionality to handle those
+* For directories and files that require special attention (Such as a requirement for an app data directory to persist after uninstall) vanagon cannot handle them
+* Vanagon cannot handle the service files required by WiX for an MSI
+* Vanagon cannot handle the localization engine that MSIs use
+
+We deal with these special cases by allowing the user to provide their own WiX files inside the vanagon project. This way the user can use the WiX toolset to solve problems like registry searching and service files.
+
+It’s possible many of these edge cases can be remedied in future releases of vanagon, such as the special casing of directories and files. But the benefit of “consistency” may not be enough to drive change for other cases, specifically because for things like custom actions the changes to vanagon would be so drastic and edge cased for windows, vanagon would start to look inconsistent with itself.
+
+### Vanagon classes, functions, and WiX
+We made no major changes to the overall class structure in vanagon, most of the work has been done in the class functions to incorporate WiX. Following along the Vanagon::Driver class function “run” :
+
+1. Install_build_dependancies remains unchanged. The underlying changes required to get dependencies for windows lies in the actual project, not vanagon itself.
+2. Fetch_sources also remains unchanged, though take note that this is NOT where we fetch any of the WiX source files. However this is where we will grab any .bat files to be put on the user’s PATH on install.
+3. Make_makefile is where some of the WiX magic begins to happen. Specifically the WiX compiler invocation exists under the generate_package function of Vanagon::Platform::Windows
+4. Make_bill_of_materials remains unchanged
+6. Generate_pacakging_artifacts is where we are actually adding / generating WiX files for the compiler to use. If you follow the objects down eventually you’ll find the generate_msi_packaging_artifacts function inside Vanagon::Platform::Windows, where you’ll notice we do a mix of taking default files in vanagon and files from the project source and putting them in the workdir.
+7. - 10. The remainder of the driver functionality remains unchanged
```

Comment:
This is where a diagram would really help - way too much tribal knowledge in getting to the stage where you and I are now ;)